### PR TITLE
DF-4019: Update Jenkins Plugin Connection Test

### PR DESCRIPTION
**Pull request by mrinehart-r7**
### Description

Describe the proposed changes:

  - Update Jenkins plugin connection test to new style. Internal customer claims it was not working properly in an orchestrator with:
```
failed to send connection test to plugin
```

### Testing

```
⇒  ./run.sh -c test build_info < tests/build_info.json
Running:  docker run --rm   -i rapid7/jenkins:1.1.2  test
Connect: Connecting...
rapid7/Jenkins:1.1.2. Step name: build_info
Starting new HTTPS connection (1): jenkins.mono-project.com:443
https://jenkins.mono-project.com:443 "GET /crumbIssuer/api/json HTTP/1.1" 404 335
https://jenkins.mono-project.com:443 "GET /job/mono-cleanup/67035/api/json?depth=0 HTTP/1.1" 200 494
{"body": {"log": "Connect: Connecting...\nrapid7/Jenkins:1.1.2. Step name: build_info\n", "status": "ok", "meta": {}, "output": {"success": true}}, "version": "v1", "type": "action_event"}
```

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

File: jenkins/help.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ The connection configuration accepts the following parameters:
 |Name|Type|Default|Required|Description|Enum|
 |----|----|-------|--------|-----------|----|
 |credentials|credential_username_password|None|True|Username and password|None|
-|host|string|None|True|Jenkins server URL|None|
+|host|string|None|True|Jenkins' server URL|None|
```
Comment (jschipp-r7) [inline jenkins/help.md]:
```suggestion
|host|string|None|True|Jenkins server URL|None|
```

Comment (mrinehart-r7) [reply jenkins/help.md]: Good catch

Comment (jschipp-r7) [reply jenkins/help.md]: Wouldn't this be like `Github server URL`?

File: jenkins/tests/build_info.json [hunk 2]
```diff
@@ -3,14 +3,14 @@
         "action": "build_info",
         "connection": {
             "credentials": {
-                "password": null,
-                "username": null
+                "password": "admin",
```
Comment (jschipp-r7) [inline jenkins/tests/build_info.json]: Clear these

**Pull request merged**